CI fails with `AttributeError: module 'lib' has no attribute 'X509_V_FLAG_NOTIFY_POLICY'. Did you mean: 'X509_V_FLAG_EXPLICIT_POLICY'?`

CI fails across the board with this error.

Run: https://github.com/coiled/benchmarks/actions/runs/12060241187/job/33630302482

Traceback: 

```
Run pytest --benchmark -n 4 --dist loadscope -m 'not tpch_nondask' tests --ignore=tests/tpch --ignore=tests/geospatial
ImportError while loading conftest '/home/runner/work/benchmarks/benchmarks/tests/conftest.py'.
tests/conftest.py:17: in <module>
    import adlfs
../../../miniconda3/envs/test/lib/python3.10/site-packages/adlfs/__init__.py:1: in <module>
    from .gen1 import AzureDatalakeFileSystem
../../../miniconda3/envs/test/lib/python3.10/site-packages/adlfs/gen1.py:9: in <module>
    from azure.datalake.store import AzureDLFileSystem, lib
../../../miniconda3/envs/test/lib/python3.10/site-packages/azure/datalake/store/__init__.py:11: in <module>
    from .core import AzureDLFileSystem
../../../miniconda3/envs/test/lib/python3.10/site-packages/azure/datalake/store/core.py:[27](https://github.com/coiled/benchmarks/actions/runs/12060241187/job/33630302482#step:13:28): in <module>
    from .lib import DatalakeRESTInterface
../../../miniconda3/envs/test/lib/python3.10/site-packages/azure/datalake/store/lib.py:39: in <module>
    from requests.packages.urllib3.contrib.pyopenssl import extract_from_urllib3 as enforce_no_py_open_ssl
../../../miniconda3/envs/test/lib/python3.10/site-packages/urllib3/contrib/pyopenssl.py:43: in <module>
    import OpenSSL.SSL  # type: ignore[import-untyped]
../../../miniconda3/envs/test/lib/python3.10/site-packages/OpenSSL/__init__.py:8: in <module>
    from OpenSSL import crypto, SSL
../../../miniconda3/envs/test/lib/python3.10/site-packages/OpenSSL/crypto.py:1579: in <module>
    class X509StoreFlags(object):
../../../miniconda3/envs/test/lib/python3.10/site-packages/OpenSSL/crypto.py:1598: in X509StoreFlags
    NOTIFY_POLICY = _lib.X509_V_FLAG_NOTIFY_POLICY
E   AttributeError: module 'lib' has no attribute 'X509_V_FLAG_NOTIFY_POLICY'. Did you mean: 'X509_V_FLAG_EXPLICIT_POLICY'?
Error: Process completed with exit code 4.
```

--- a/AB_environments/AB_baseline.conda.yaml
+++ b/AB_environments/AB_baseline.conda.yaml
@@ -58,7 +58,7 @@ dependencies:
   - adlfs ==2024.7.0
   # https://github.com/coiled/benchmarks/issues/1616
   - cryptography ==44.0.0
-  - pyopenssl ==24.3.0
+  - pyopenssl ==24.2.1
   # End copy-paste
 
   - pip:

--- a/AB_environments/AB_baseline.conda.yaml
+++ b/AB_environments/AB_baseline.conda.yaml
@@ -10,54 +10,54 @@ dependencies:
   # Copy-paste from ci/environment.yml
   - pip
   - coiled >=0.2.54
-  - numpy ==2.0
-  - pandas ==2.2.2
-  - dask ==2024-07.1
-  - distributed ==2024-07.1
-  - dask-expr ==1.1.9
+  - numpy ==2.0.2
+  - pandas ==2.2.3
+  - dask ==2024.11.2
+  - distributed ==2024.11.2
+  - dask-expr ==1.1.19
   - dask-labextension ==7.0.0
   - dask-ml ==2024.4.4
-  - fsspec ==2024.6.1
-  - s3fs ==2024.6.1
-  - gcsfs ==2024.6.1
-  - pyarrow ==18.0.0
-  - jupyterlab ==4.2.4
+  - fsspec ==2024.10.0
+  - s3fs ==2024.10.0
+  - gcsfs ==2024.10.0
+  - pyarrow ==18.1.0
+  - jupyterlab ==4.3.1
   - lz4 ==4.3.3
-  - ipywidgets ==8.1.3
+  - ipywidgets ==8.1.5
   - numba ==0.60.0
-  - scikit-learn ==1.5.1
+  - scikit-learn ==1.5.2
   - ipycytoscape ==1.3.3
   - click ==8.1.7
-  - xarray ==2024.07.0
-  - flox ==0.9.9
-  - zarr ==2.18.2
+  - xarray ==2024.11.0
+  - flox ==0.9.15
+  - zarr ==2.18.3
   - cftime ==1.6.4
   - msgpack-python
-  - cloudpickle ==3.0.0
-  - tornado ==6.4.1
-  - toolz ==0.12.1
+  - cloudpickle ==3.1.0
+  - tornado ==6.4.2
+  - toolz ==1.0.0
   - zict ==3.0.0
-  - xgboost ==2.1.1
-  - optuna ==3.6.1
-  - optuna-integration ==3.6.0
-  - scipy ==1.14.0
-  - sqlalchemy ==2.0.31
+  - xgboost ==2.1.2
+  - optuna ==4.1.0
+  - optuna-integration ==4.1.0
+  - scipy ==1.14.1
+  - sqlalchemy ==2.0.36
   - pynvml ==11.5.3
-  - bokeh ==3.5.1
+  - bokeh ==3.6.1
   - gilknocker ==0.4.1
   - openssl >1.1.0g
   - rasterio >=1.4.0
   - rioxarray ==0.17.0
-  - h5netcdf ==1.3.0
+  - h5netcdf ==1.4.1
   - xesmf ==0.8.7
-  - bottleneck ==1.4.1
+  - bottleneck ==1.4.2
   - geojson ==3.1.0
   - planetary-computer ==1.0.0
-  - pystac-client ==0.8.3
+  - pystac-client ==0.8.5
   - odc-stac ==0.3.10
   - adlfs ==2024.7.0
   # https://github.com/coiled/benchmarks/issues/1616
-  - cryptography ==44.0.0
+  - cryptography ==43.0.3
   - pyopenssl ==24.2.1
   # End copy-paste
 

--- a/AB_environments/AB_baseline.conda.yaml
+++ b/AB_environments/AB_baseline.conda.yaml
@@ -6,6 +6,7 @@ channels:
   - conda-forge
 dependencies:
   - python =3.10  # Single '=' means latest patch version available
+  - memray ==1.13.4
   # Copy-paste from ci/environment.yml
   - pip
   - coiled >=0.2.54
@@ -45,8 +46,8 @@ dependencies:
   - bokeh ==3.5.1
   - gilknocker ==0.4.1
   - openssl >1.1.0g
-  - rioxarray ==0.17.0
   - rasterio >=1.4.0
+  - rioxarray ==0.17.0
   - h5netcdf ==1.3.0
   - xesmf ==0.8.7
   - bottleneck ==1.4.1
@@ -55,7 +56,9 @@ dependencies:
   - pystac-client ==0.8.3
   - odc-stac ==0.3.10
   - adlfs ==2024.7.0
-  - memray ==1.13.4
+  # https://github.com/coiled/benchmarks/issues/1616
+  - cryptography ==44.0.0
+  - pyopenssl ==24.3.0
   # End copy-paste
 
   - pip:

--- a/AB_environments/AB_sample.conda.yaml
+++ b/AB_environments/AB_sample.conda.yaml
@@ -64,7 +64,7 @@ dependencies:
   - adlfs ==2024.7.0
   # https://github.com/coiled/benchmarks/issues/1616
   - cryptography ==44.0.0
-  - pyopenssl ==24.3.0
+  - pyopenssl ==24.2.1
   # End copy-paste
 
   - pip:

--- a/AB_environments/AB_sample.conda.yaml
+++ b/AB_environments/AB_sample.conda.yaml
@@ -12,6 +12,7 @@ channels:
   - conda-forge
 dependencies:
   - python =3.10  # Single '=' means latest patch version available
+  - memray ==1.13.4
   # Copy-paste from ci/environment.yml
   - pip
   - coiled >=0.2.54
@@ -51,8 +52,8 @@ dependencies:
   - bokeh ==3.5.1
   - gilknocker ==0.4.1
   - openssl >1.1.0g
-  - rioxarray ==0.17.0
   - rasterio >=1.4.0
+  - rioxarray ==0.17.0
   - h5netcdf ==1.3.0
   - xesmf ==0.8.7
   - bottleneck ==1.4.1
@@ -61,7 +62,9 @@ dependencies:
   - pystac-client ==0.8.3
   - odc-stac ==0.3.10
   - adlfs ==2024.7.0
-  - memray ==1.13.4
+  # https://github.com/coiled/benchmarks/issues/1616
+  - cryptography ==44.0.0
+  - pyopenssl ==24.3.0
   # End copy-paste
 
   - pip:

--- a/AB_environments/AB_sample.conda.yaml
+++ b/AB_environments/AB_sample.conda.yaml
@@ -16,54 +16,54 @@ dependencies:
   # Copy-paste from ci/environment.yml
   - pip
   - coiled >=0.2.54
-  - numpy ==2.0
-  - pandas ==2.2.2
-  - dask ==2024-07.1
-  - distributed ==2024-07.1
-  - dask-expr ==1.1.9
+  - numpy ==2.0.2
+  - pandas ==2.2.3
+  - dask ==2024.11.2
+  - distributed ==2024.11.2
+  - dask-expr ==1.1.19
   - dask-labextension ==7.0.0
   - dask-ml ==2024.4.4
-  - fsspec ==2024.6.1
-  - s3fs ==2024.6.1
-  - gcsfs ==2024.6.1
-  - pyarrow ==18.0.0
-  - jupyterlab ==4.2.4
+  - fsspec ==2024.10.0
+  - s3fs ==2024.10.0
+  - gcsfs ==2024.10.0
+  - pyarrow ==18.1.0
+  - jupyterlab ==4.3.1
   - lz4 ==4.3.3
-  - ipywidgets ==8.1.3
+  - ipywidgets ==8.1.5
   - numba ==0.60.0
-  - scikit-learn ==1.5.1
+  - scikit-learn ==1.5.2
   - ipycytoscape ==1.3.3
   - click ==8.1.7
-  - xarray ==2024.07.0
-  - flox ==0.9.9
-  - zarr ==2.18.2
+  - xarray ==2024.11.0
+  - flox ==0.9.15
+  - zarr ==2.18.3
   - cftime ==1.6.4
   - msgpack-python
-  - cloudpickle ==3.0.0
-  - tornado ==6.4.1
-  - toolz ==0.12.1
+  - cloudpickle ==3.1.0
+  - tornado ==6.4.2
+  - toolz ==1.0.0
   - zict ==3.0.0
-  - xgboost ==2.1.1
-  - optuna ==3.6.1
-  - optuna-integration ==3.6.0
-  - scipy ==1.14.0
-  - sqlalchemy ==2.0.31
+  - xgboost ==2.1.2
+  - optuna ==4.1.0
+  - optuna-integration ==4.1.0
+  - scipy ==1.14.1
+  - sqlalchemy ==2.0.36
   - pynvml ==11.5.3
-  - bokeh ==3.5.1
+  - bokeh ==3.6.1
   - gilknocker ==0.4.1
   - openssl >1.1.0g
   - rasterio >=1.4.0
   - rioxarray ==0.17.0
-  - h5netcdf ==1.3.0
+  - h5netcdf ==1.4.1
   - xesmf ==0.8.7
-  - bottleneck ==1.4.1
+  - bottleneck ==1.4.2
   - geojson ==3.1.0
   - planetary-computer ==1.0.0
-  - pystac-client ==0.8.3
+  - pystac-client ==0.8.5
   - odc-stac ==0.3.10
   - adlfs ==2024.7.0
   # https://github.com/coiled/benchmarks/issues/1616
-  - cryptography ==44.0.0
+  - cryptography ==43.0.3
   - pyopenssl ==24.2.1
   # End copy-paste
 

--- a/ci/environment.yml
+++ b/ci/environment.yml
@@ -57,6 +57,9 @@ dependencies:
   - pystac-client ==0.8.3
   - odc-stac ==0.3.10
   - adlfs ==2024.7.0
+  # https://github.com/coiled/benchmarks/issues/1616
+  - cryptography ==44.0.0
+  - pyopenssl ==24.3.0
 
 ########################################################
 # PLEASE READ:

--- a/ci/environment.yml
+++ b/ci/environment.yml
@@ -59,7 +59,7 @@ dependencies:
   - adlfs ==2024.7.0
   # https://github.com/coiled/benchmarks/issues/1616
   - cryptography ==44.0.0
-  - pyopenssl ==24.3.0
+  - pyopenssl ==24.2.1
 
 ########################################################
 # PLEASE READ:

--- a/ci/environment.yml
+++ b/ci/environment.yml
@@ -11,54 +11,54 @@ dependencies:
   - python >=3.10
   - pip
   - coiled >=0.2.54
-  - numpy ==2.0
-  - pandas ==2.2.2
-  - dask ==2024-07.1
-  - distributed ==2024-07.1
-  - dask-expr ==1.1.9
+  - numpy ==2.0.2
+  - pandas ==2.2.3
+  - dask ==2024.11.2
+  - distributed ==2024.11.2
+  - dask-expr ==1.1.19
   - dask-labextension ==7.0.0
   - dask-ml ==2024.4.4
-  - fsspec ==2024.6.1
-  - s3fs ==2024.6.1
-  - gcsfs ==2024.6.1
-  - pyarrow ==18.0.0
-  - jupyterlab ==4.2.4
+  - fsspec ==2024.10.0
+  - s3fs ==2024.10.0
+  - gcsfs ==2024.10.0
+  - pyarrow ==18.1.0
+  - jupyterlab ==4.3.1
   - lz4 ==4.3.3
-  - ipywidgets ==8.1.3
+  - ipywidgets ==8.1.5
   - numba ==0.60.0
-  - scikit-learn ==1.5.1
+  - scikit-learn ==1.5.2
   - ipycytoscape ==1.3.3
   - click ==8.1.7
-  - xarray ==2024.07.0
-  - flox ==0.9.9
-  - zarr ==2.18.2
+  - xarray ==2024.11.0
+  - flox ==0.9.15
+  - zarr ==2.18.3
   - cftime ==1.6.4
   - msgpack-python
-  - cloudpickle ==3.0.0
-  - tornado ==6.4.1
-  - toolz ==0.12.1
+  - cloudpickle ==3.1.0
+  - tornado ==6.4.2
+  - toolz ==1.0.0
   - zict ==3.0.0
-  - xgboost ==2.1.1
-  - optuna ==3.6.1
-  - optuna-integration ==3.6.0
-  - scipy ==1.14.0
-  - sqlalchemy ==2.0.31
+  - xgboost ==2.1.2
+  - optuna ==4.1.0
+  - optuna-integration ==4.1.0
+  - scipy ==1.14.1
+  - sqlalchemy ==2.0.36
   - pynvml ==11.5.3
-  - bokeh ==3.5.1
+  - bokeh ==3.6.1
   - gilknocker ==0.4.1
   - openssl >1.1.0g
   - rasterio >=1.4.0
   - rioxarray ==0.17.0
-  - h5netcdf ==1.3.0
-  - xesmf ==0.8.7
-  - bottleneck ==1.4.1
+  - h5netcdf ==1.4.1
+  - xesmf ==0.8.8
+  - bottleneck ==1.4.2
   - geojson ==3.1.0
   - planetary-computer ==1.0.0
-  - pystac-client ==0.8.3
+  - pystac-client ==0.8.5
   - odc-stac ==0.3.10
   - adlfs ==2024.7.0
   # https://github.com/coiled/benchmarks/issues/1616
-  - cryptography ==44.0.0
+  - cryptography ==43.0.3
   - pyopenssl ==24.2.1
 
 ########################################################

--- a/ci/environment.yml
+++ b/ci/environment.yml
@@ -50,7 +50,7 @@ dependencies:
   - rasterio >=1.4.0
   - rioxarray ==0.17.0
   - h5netcdf ==1.4.1
-  - xesmf ==0.8.8
+  - xesmf ==0.8.7
   - bottleneck ==1.4.2
   - geojson ==3.1.0
   - planetary-computer ==1.0.0

